### PR TITLE
cmd/xdpderper: add autodetection for default interface name

### PR DESCRIPTION
**Pull request by raggi**
This makes deployment easier in hetrogenous environments.

Updates ENG-4274

### CODE REVIEW

File: cmd/xdpderper/xdpderper.go [hunk 1]
```diff
@@ -41,8 +42,17 @@ func main() {
 	default:
 		log.Fatal("invalid mode")
 	}
+	deviceName := *flagDevice
+	if deviceName == "" {
+		var err error
+		deviceName, _, err = netutil.DefaultInterfacePortable()
+		if err != nil || deviceName == "" {
+			log.Fatalf("failed to detect default route interface: %v", err)
+		}
```
Comment (jwhited) [inline cmd/xdpderper/xdpderper.go:51]: should we log what was autodetected here? or perhaps as part of the existing log message on L68 just always log the device name being attached to?

File: net/netutil/default_interface_portable.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+)
+
+// DefaultInterfacePortable looks up the current default interface using a portable lookup method that
+// works on most systems with a BSD style socket interface.
+//
+// Returns the interface name and IP address of the default route interface.
+//
+// If the default cannot be determined, an error is returned.
+// Requires that there is a route on the system servicing UDP IPv4.
+func DefaultInterfacePortable() (string, netip.Addr, error) {
```
Comment (jwhited) [inline net/netutil/default_interface_portable.go:16]: Do we have something similar already in the tree? I have a vague memory of such a thing, but could be wrong.

**Review by jwhited — APPROVED** [review]

File: net/netutil/default_interface_portable.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
+package netutil
```
Comment (jwhited) [inline net/netutil/default_interface_portable.go:1]: license

File: net/netutil/default_interface_portable_test.go [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package netutil
```
Comment (jwhited) [inline net/netutil/default_interface_portable_test.go:1]: license

File: net/netutil/default_interface_portable.go (hunk 2 above)
Comment (raggi) [reply net/netutil/default_interface_portable.go:19]: It used to be in net/interfaces, but it was merged into netmon which feels too dynamic & heavy for this purpose. The approach that they take is also quite complex, and I've been meaning to type this approach into some code somewhere for a while, as it's extremely useful and portable compared to parsing the routing tables as done elsewhere.

**Pull request merged**